contracts/proxy-lib/src/lib.rs

```diff
@@ -0,0 +1,408 @@
+use core::str;
+use std::collections::HashSet;
+use std::usize;
+
+use calimero_context_config::repr::{Repr, ReprTransmute};
+use calimero_context_config::types::{ContextId, Signed, SignerId};
+use near_sdk::json_types::{Base64VecU8, U128};
+use near_sdk::store::IterableMap;
+use near_sdk::{
+    env, near, AccountId, Gas, NearToken, PanicOnDefault, Promise, PromiseError, PromiseOrValue,
+    PromiseResult,
+};
+
+pub mod ext_config;
+pub use crate::ext_config::config_contract;
+
+pub type ProposalId = u32;
+
+#[derive(PartialEq, Debug)]
+#[near(serializers = [json, borsh])]
+pub struct ProposalWithApprovals {
+    pub proposal_id: ProposalId,
+    pub num_approvals: usize,
+}
+
+enum MemberAction {
+    Approve {
+        identity: Repr<SignerId>,
+        proposal_id: ProposalId,
+    },
+    Create {
+        proposal: Proposal,
+        num_proposals: u32,
+    },
+}
+#[near(contract_state)]
+#[derive(PanicOnDefault)]
+pub struct ProxyContract {
+    pub context_id: ContextId,
+    pub context_config_account_id: AccountId,
+    pub num_approvals: u32,
+    pub proposal_nonce: ProposalId,
+    pub proposals: IterableMap<ProposalId, Proposal>,
+    pub approvals: IterableMap<ProposalId, HashSet<SignerId>>,
```

Comment:
_... if this was a `BTreeSet`

```suggestion
    pub approvals: IterableMap<ProposalId, BTreeSet<SignerId>>,
```

are you concerned about rebalancing on insertion since `SignerId` is random? 32-bytes is pretty small comparison